c-ext/decompressor.c

```diff
@@ -299,15 +299,15 @@ PyObject *Decompressor_decompress(ZstdDecompressor *self, PyObject *args,
         goto finally;
     }
 
-    decompressedSize = ZSTD_getFrameContentSize(source.buf, source.len);
-
-    if (ZSTD_CONTENTSIZE_ERROR == decompressedSize) {
+    ZSTD_frameHeader zfh;
+    if (ZSTD_getFrameHeader_advanced(&zfh, source.buf, source.len, self->format) != 0) {
         PyErr_SetString(ZstdError,
                         "error determining content size from frame header");
         goto finally;
     }
+    decompressedSize=zfh.frameContentSize;
```

Comment:
Will inferring from the frame's size be incompatible with `readAcrossFrames`?

Reply:
That doesn't seem to implemented anyways... https://github.com/swehner/python-zstandard/blob/main/c-ext/decompressor.c#L293 (anyways i think this should be equivalent to the previous code, shouldn't it?)